### PR TITLE
collect email on demo project

### PR DESCRIPTION
**Pull request by Vadman97**
## Summary

Collects emails on the demo project.
Gates the project from loading until the email is provided.
Reports the email as an H.identify property and as a track event that we can load.

## How did you test this change?

Screenshot from 2023-10-24 14-46-34

Screenshot from 2023-10-24 15-56-26

## Are there any deployment considerations?

No

## Does this work require review from our design team?

yes @julian-highlight

### CODE REVIEW

File: backend/integrations/github/github_test.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package github
+
+import (
+	"context"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestGetTree(t *testing.T) {
```
Comment (Vadman97) [inline backend/integrations/github/github_test.go:9]: unrelated to this pr, but adding a new test to check our github integration

**Review by julian-highlight — APPROVED** [review]
Looks good to me!

**Review by SpennyNDaJets — APPROVED** [review]
Looks good! Only product question - did we intend to gate behind email, even when user is logged in?

**Comment by Vadman97** [comment]
> Looks good! Only product question - did we intend to gate behind email, even when user is logged in?

good catch @SpennyNDaJets , i'll override the gate when logged in :pray:

**Pull request merged**